### DISCUSSION TITLE
ssh:// URLs in custom fields

### DISCUSSION THREAD

**Discussion by n0i2** [netbox-community/netbox#8313] [Ideas]
Greetings!

In previous Netbox versions it was allowed to use ssh:// URLs in custom fields. When trying to add such a link in version 3.1 you get an error message saying "Enter a valid URL".

Any chance that you allow the use of ssh:// URLs again?

Thanks in advance!

**Comment by candlerb** [comment]
Netbox's EnhancedURLValidator is based on Django's [URLValidator](https://docs.djangoproject.com/en/4.0/ref/validators/#urlvalidator).  Unless you supply an explicit list of schemes - which as far as I can see [isn't given](https://github.com/netbox-community/netbox/blob/v3.1.5/netbox/utilities/forms/fields.py#L112) - then it accepts only `['http', 'https', 'ftp', 'ftps']`

I'd say it's a reasonable feature request to be able to provide a list, e.g. in the config file; even better would be at the field level.

**Reply by jeremystretch** [reply]
This is already a configurable setting: https://netbox.readthedocs.io/en/stable/configuration/dynamic-settings/#allowed_url_schemes

**Reply by candlerb** [reply]
Don't know how I missed that.  Thanks!

**Reply by candlerb** [reply]
Hmm, the Netbox default should include `ssh`, but I've tested it and it doesn't work.  Bug opened at #8319
